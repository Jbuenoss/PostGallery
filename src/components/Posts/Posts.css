.posts-container {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(280px, 1fr)); /* Ajusta automaticamente o número de colunas */
    gap: 20px; /* Espaçamento entre os posts */
}