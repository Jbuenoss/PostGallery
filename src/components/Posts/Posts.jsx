import { PostCard } from "../PostCard/PostCard.jsx";
import './Posts.css';

// Initialize allPosts as an empty array to avoid map on undefined
export const Posts = ({ allposts = [] }) => (
    <div className="posts-container">
        {allposts.map((post) => (
            <PostCard
                key={post.id}
                post={post} />
        ))}
    </div>
);