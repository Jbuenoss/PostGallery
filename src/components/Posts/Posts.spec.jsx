import { Posts } from './Posts.jsx';
import { render, screen } from '@testing-library/react';

const props = {
    posts: [
        {
            id: 0,
            title: 'title 0',
            body: 'body 0',
            cover: 'img0.png'
        },
        {
            id: 1,
            title: 'title 1',
            body: 'body 1',
            cover: 'img1.png'
        }
    ]
}

describe('<Posts />', () => {
    it('should render posts', () => {
        render(<Posts allposts={props.posts} />);

        expect(screen.getAllByRole('heading', {name: /title/i}))
            .toHaveLength(2);
        expect(screen.getAllByRole('img', {name: /title/i}))
            .toHaveLength(2);
        expect(screen.getAllByText(/body/i))
            .toHaveLength(2);
    });

    it('should not render posts', () => {
        render(<Posts />);

        expect(screen.queryByRole('heading', {name: /title/i}))
            .not.toBeInTheDocument();
        
    });

    it('should match snapshot', () => {
        const {container} = render(<Posts allposts={props.posts} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});