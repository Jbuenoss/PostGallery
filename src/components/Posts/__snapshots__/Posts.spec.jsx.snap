// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts-container"
>
  <div
    class="post"
  >
    <img
      alt="title 0"
      src="img0.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 0
      </h2>
      <p>
        body 0
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
</div>
`;
